pkg/apis/listeners/v1alpha1/listenerclass_types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2024 zncdatadev.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ListenerClassSpec defines the desired state of ListenerClass
+type ListenerClassSpec struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=LoadBalancer;NodePort;ClusterIP
+	ServiceType *corev1.ServiceType `json:"serviceType,omitempty"`
```

Comment:
Consider using direct value assignment for `ServiceType` instead of a pointer to avoid unnecessary nil checks and improve code clarity. This change could enhance the efficiency of the code by leveraging the default zero value of the `ServiceType` enum if applicable. [important]